### ISSUE TITLE
Submit forms to the current URL if [action] attribute is given, but set to an empty string

### ISSUE THREAD

**Issue by j4mie** [unpoly/unpoly#248]
### Bug description

When using `[up-validate]`, an exception is raised.

### Reproduction project

https://responsible-simple-rise.glitch.me

### Steps to reproduce the behavior:

1. Go to reproduction project linked above
2. Open the dev tools panel
3. Enter something in the form field
4. Hit tab
5. See error in dev tools panel:

<img width="716" alt="image" src="https://user-images.githubusercontent.com/6988/128680127-f5a158e2-ccaa-480f-a7de-a4b587933122.png">

### Expected behavior

`[up-validate]` should behave as described in the docs.

### Browser version

 - OS: macOS
 - Browser: Firefox
 - Version: 91.0b7

### Additional context

I've tried various values for `[up-validate]` and it seems to fail whatever I do.

**Comment by j4mie** [comment]
Update: it turns out this was because the form's `action` attribute was empty (`""`). This generally seems to work in browsers but is technically a violation of the spec - see https://stackoverflow.com/a/1132015

Happy to close this issue if you think the above explains it sufficiently!

**Comment by adam12** [comment]
> Happy to close this issue if you think the above explains it sufficiently!

This might be something we want to warn against, so I'd say let's keep it open for now.

> form's `action` attribute was empty

Was this the original configuration that you saw the exception in? If so, was that an oversight or was it intentional? Curious to know what `up-validate` would be submitting to if it was intentional.

**Comment by j4mie** [comment]
> Was this the original configuration that you saw the exception in? If so, was that an oversight or was it intentional? Curious to know what `up-validate` would be submitting to if it was intentional.

It was intentional in the sense that my understanding of `action=""` was "submit this form to the same URL that you loaded it from". It turns out that this behaviour happens to work in most browsers, but the correct way to express "submit this form to the same URL that you loaded it from" is just to omit the `action=` attribute entirely.

My hunch is that Unpoly should treat `action=""` in the same way it treats a form without an action attribute, and submit the form to the same URL that the browser is currently on.

**Comment by adam12** [comment]
Interesting. I've never intentionally used an empty `action`, as I usually just omit it.

I think you're right tho, and Unpoly should fallback in this case to duplicate browser behaviour.

**Comment by adam12** [comment]
The original Glitch is 404, but I [tried recreating it here](https://glitch.com/edit/#!/different-bevel-beginner).

When causing the input to blur, I noticed that Unpoly will submit a validation request to the current URL with the default form submission method (GET), and load the correct fragment.

Confirmed with:

Chrome Version 113.0.5672.126 (Official Build) (arm64)
Firefox 113.0.2 (64-bit)
Safari Version 16.5 (18615.2.9.11.4)

So I'm going to mark this as closed.
